server/server/api/views/views.py

```diff
@@ -669,147 +670,60 @@ def get(self):
 
 class Donations(Resource):
     def post(self):
-        """
-        Process the payment
-        ---
-        responses:
-            all:
-                description: Information massage.
-                type: string
-        """
-        data = json.loads(list(request.form.keys())[0])
-        currency = data.get('currency')
-        inputted_amount = data.get('amount')
-        one_time_donation = data.get('oneTimeDonation')
-        monthly_donation = data.get('monthlyDonation')
-        stripe_data = data.get('stripe')
-        name = data.get('name')
-        email_address = data.get('email')
-        message = data.get('message')
-
-        secret_key = os.environ.get('STRIPE_SECRET')
-
-        stripe.api_key = secret_key
-        db = get_db()
-        try:
-            currency = list(db['currencies'].find({'symbol': currency}))[0]
-        except IndexError:
-            return 'No such currency', 400
-
-        if currency['decimal']:
-            amount = inputted_amount * 100
-        else:
-            amount = inputted_amount
-
-        customer_data = {'source': stripe_data['id']}
-
-        if email_address:
-            customer_data['email'] = email_address
-
-        try:
-            customer = stripe.Customer.create(**customer_data)
-        except stripe.CardError:
-            return {'err_code': 3}, 400
-        try:
-            if one_time_donation:
-                charge = stripe.Charge.create(
-                    customer=customer.id,
-                    amount=amount,
-                    currency=currency['symbol'],
-                    metadata={"name": name, "message": message},
-                    description=f'''Donation by {name if name else ""}, 
-                                message {message if message else ""}''',
+        body = request.get_json()
+        if body is not None and all(item in list(body.keys()) for item in ['currency', 'amount', 'frequency']):
+            currency = body['currency']
+            amount = body['amount']
+            frequency = body['frequency']
+
+            secret_key = os.environ.get('STRIPE_SECRET')
+            stripe.api_key = secret_key
+
+            incoming_uri = urlparse(request.url)
+            cancel_url = '{uri.scheme}://{uri.netloc}/donate-now'.format(uri=incoming_uri)
+            success_url = '{uri.scheme}://{uri.netloc}/thank-you'.format(uri=incoming_uri)
```

Comment:
Hmm. Is `uri.netloc` validated somehow? I.e. like Django does "ALLOWED_HOSTS"? Otherwise this is susceptible of header poisoning.

Reply:
tbh I don't know. Should we really bother it? It's only plain link. If it's a problem, it should be fix in separate PR.

Reply:
Maybe I can take the domain form some kind of configuration. I'll check it.